File: src/Vincent OS/Terminal.vb
Public Class Terminal

    Private Sub Terminal_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        If Bureau.Visible = True Then
            Bureau.Button24.Visible = True
        ElseIf Bureau2.Visible = True Then
            Bureau2.Button24.Visible = True
        End If
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        '' Création d'un Private Sub pour mieux se retrouver
        commande()
    End Sub

    Private Sub TextBox1_KeyPress(ByVal sender As System.Object, ByVal e As System.Windows.Forms.KeyPressEventArgs) Handles TextBox1.KeyPress
        '' Faire en sorte que la touche Entrée puisse marcher
        If e.KeyChar = Convert.ToChar(Keys.Enter) Then
            commande()
        End If
    End Sub

    Private Sub Terminal_FormClosed(ByVal sender As System.Object, ByVal e As System.Windows.Forms.FormClosedEventArgs) Handles MyBase.FormClosed
        If Bureau.Visible = True Then
            Bureau.Button24.Visible = False
        ElseIf Bureau2.Visible = True Then
            Bureau2.Button24.Visible = False
        End If
    End Sub

    Private Sub commande()
        If TextBox1.Text = "ver" Then
            ''ATTENTION A BIEN CHANGER LA VERSION CAR JE VAIS SUREMENT OUBLIER A LA PROCHAINE VERSION !!!!!!!!!!!
            RichTextBox1.Text = "Nom du système d'exploitation : Vincent OS" & vbCrLf & "Version : 7.1" & vbCrLf & "Branche : Beta"
            TextBox1.Text = ""
        ElseIf TextBox1.Text = "help" Then
            RichTextBox1.Text = "Bing Chrome        Lance Bing Chrome." & vbCrLf & "Calc       Lance la calculatrice" & vbCrLf & "Chrono       Lance le chronomètre" & vbCrLf & "Exewin       Lance l'exécuteur d'application non Vincent OS." & vbCrLf & "Game        Lance le gestionnaire de jeux." & vbCrLf & "MediaPlayer      Lance le Media Player." & vbCrLf & "Meme        Lance l'explorateur de fichiers." & vbCrLf & "Notepad        Lance le Bloc Note." & vbCrLf & "Paint       Lance Paint." & vbCrLf & "Store        Lance Vincent OS Store." & vbCrLf & "Terminal     Lance une autre fenêtre d'un Terminal." & vbCrLf & "Visio       Lance la visionneuse d'images." & vbCrLf & vbCrLf & "Certaines commandes sont cachées !"
            TextBox1.Text = ""
        ElseIf TextBox1.Text = "Bing Chrome" Then
            Bing_Chrome.Show()
            TextBox1.Text = ""
        ElseIf TextBox1.Text = "Calc" Then
            Calc.Show()
            TextBox1.Text = ""
        ElseIf TextBox1.Text = "Exewin" Then
            exewin.Show()
            TextBox1.Text = ""
        ElseIf TextBox1.Text = "Game" Then
            game.Show()
            TextBox1.Text = ""
        ElseIf TextBox1.Text = "MediaPlayer" Then
            Media.Show()
            TextBox1.Text = ""
        ElseIf TextBox1.Text = "Meme" Then
            meme.Show()
            TextBox1.Text = ""
        ElseIf TextBox1.Text = "Notepad" Then
            Notepad.Show()
            TextBox1.Text = ""
        ElseIf TextBox1.Text = "Paint" Then
            paints.Show()
            TextBox1.Text = ""
        ElseIf TextBox1.Text = "Store" Then
            Store.Show()
            TextBox1.Text = ""
        ElseIf TextBox1.Text = "Terminal" Then
            Dim cp As New Terminal
            cp.Show()
            TextBox1.Text = ""
        ElseIf TextBox1.Text = "Visio" Then
            Visio.Show()
            TextBox1.Text = ""
            '' Applications disponible sur le store
        ElseIf TextBox1.Text = "Chrono" Then
            If Applications.Button11.Visible = True Then
                chrono.Show()
            Else
                RichTextBox1.Text = "ERREUR : Cette application n'est pas installé."
            End If
            TextBox1.Text = ""
            ''Ici commence les commandes de Debug, à utiliser avec précaution !

        ElseIf TextBox1.Text = "Debug_Options" Then
            RichTextBox1.Text = "DEBUG_ERREUR : Veuillez rentrer une sous option" & vbCrLf & "Pour savoir les différentes commandes, tapez : Debug_Options_ Help"
            TextBox1.Text = ""
        ElseIf TextBox1.Text = "Debug_Options_Custom" Then
            If Bureau.Visible = True Or Bureau2.Visible = True Then
                Custom.Show()
            Else
                MessageBox.Show("DEBUG_ERREUR : Il n'y a pas d'interface graphique et/ou de bureau d'ouvert", "DEBUG_ERREUR", MessageBoxButtons.OK, MessageBoxIcon.Error)
            End If
            TextBox1.Text = ""
        ElseIf TextBox1.Text = "Debug_Options_Help" Then
            RichTextBox1.Text = "Debug_Options_Custom"
            TextBox1.Text = ""
        ElseIf TextBox1.Text = "Debug_AlerteLog_ForceCrash" Then
            Application.Exit()
        ElseIf TextBox1.Text = "Debug_AlerteLog" Then
            alertelog.Show()
            TextBox1.Text = ""
        ElseIf TextBox1.Text = "Debug_Help" Then
            RichTextBox1.Text = "Debug_AlerteLog" & vbCrLf & "Debug_AlerteLog_ForceCrash" & vbCrLf & "Debug_Options"
            TextBox1.Text = ""
        Else
            RichTextBox1.Text = "Commande non reconnu."
            TextBox1.Text = ""
        End If
    End Sub
End Class

File: src/Vincent OS/Terminal.cs
using System;
using System.Windows.Forms;

namespace Vincent_OS
{
    public partial class Terminal
    {
        Bureau bureau = new Bureau();
        Bureau2 bureau2 = new Bureau2();
        public Terminal()
        {
            InitializeComponent();
        }

        private void Terminal_Load(object sender, EventArgs e)
        {
            if (bureau.Visible == true)
            {
                bureau.Button24.Visible = true;
            }
            else if (bureau2.Visible == true)
            {
                bureau2.Button24.Visible = true;
            }
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            // Création d'un private void pour mieux se retrouver
            commande();
        }

        private void TextBox1_KeyPress(object sender, KeyPressEventArgs e)
        {
            // Faire en sorte que la touche Entrée puisse marcher
            if (e.KeyChar == Convert.ToChar((int)Keys.Enter))
            {
                commande();
            }
        }

        private void Terminal_FormClosed(object sender, FormClosedEventArgs e)
        {
            if (bureau.Visible == true)
            {
                bureau.Button24.Visible = false;
            }
            else if (bureau2.Visible == true)
            {
                bureau2.Button24.Visible = false;
            }
            GC.Collect();
            GC.WaitForPendingFinalizers();
        }

        private void commande()
        {
            if (TextBox1.Text == "ver")
            {
                // ATTENTION A BIEN CHANGER LA VERSION CAR JE VAIS SUREMENT OUBLIER A LA PROCHAINE VERSION !!!!!!!!!!!
                RichTextBox1.Text = 
                    "Nom du système d'exploitation : Vincent OS\n" +
                    "Version : 10\n" +
                    "Branche : Officiel";
                TextBox1.Text = "";
            }
            else if (TextBox1.Text == "help")
            {
                RichTextBox1.Text = "Bing Chrome - Lance Bing Chrome.\n" +
                    "Calc - Lance la calculatrice\n" +
                    "Exewin - Lance l'exécuteur d'application non Vincent OS.\n" +
                    "Game - Lance le gestionnaire de jeux.\n" +
                    "MediaPlayer - Lance le Media Player.\n" +
                    "Meme - Lance l'explorateur de fichiers.\n" +
                    "Notepad - Lance le Bloc Note.\n" +
                    "Paint - Lance Paint.\n" +
                    "Terminal - Lance une autre fenêtre d'un Terminal.\n" +
                    "Visio - Lance la visionneuse d'images.\n\n" +
                    "Certaines commandes sont cachées !";
                TextBox1.Text = "";
            }
            else if (TextBox1.Text == "Bing Chrome")
            {
                WPF.Bing_Chrome bingChrome = new WPF.Bing_Chrome();
                bingChrome.Show();
                TextBox1.Text = "";
            }
            else if (TextBox1.Text == "Calc")
            {
                Calc calc = new Calc();
                calc.Show();
                TextBox1.Text = "";
            }
            else if (TextBox1.Text == "Exewin")
            {
                exewin exewin = new exewin();
                exewin.Show();
                TextBox1.Text = "";
            }
            else if (TextBox1.Text == "Game")
            {
                game game = new game();
                game.Show();
                TextBox1.Text = "";
            }
            else if (TextBox1.Text == "MediaPlayer")
            {
                WPF.Media media = new WPF.Media();
                media.Show();
                TextBox1.Text = "";
            }
            else if (TextBox1.Text == "Meme")
            {
                meme meme = new meme();
                meme.Show();
                TextBox1.Text = "";
            }
            else if (TextBox1.Text == "Notepad")
            {
                Notepad notepad = new Notepad();
                notepad.Show();
                TextBox1.Text = "";
            }
            else if (TextBox1.Text == "Paint")
            {
                paints paints = new paints();
                paints.Show();
                TextBox1.Text = "";
            }
            else if (TextBox1.Text == "Terminal")
            {
                Terminal terminal = new Terminal();
                terminal.Show();
                TextBox1.Text = "";
            }
            else if (TextBox1.Text == "Visio")
            {
                Visio visio = new Visio();
                visio.Show();
                TextBox1.Text = "";
            }
            // Ici commence les commandes de Debug, à utiliser avec précaution !

            else if (TextBox1.Text == "Debug_Options")
            {
                RichTextBox1.Text = 
                    "[DEBUG]: Enter a valid command.\n" +
                    "Debug_Options_Help";
                TextBox1.Text = "";
            }
            else if (TextBox1.Text == "Debug_Options_Custom")
            {
                if (bureau.Visible == true | bureau2.Visible == true)
                {
                    Custom custom = new Custom();
                    custom.Show();
                }
                else
                {
                    RichTextBox1.Text = "[DEBUG_ERROR]: No Bureau opened.";
                }
                TextBox1.Text = "";
            }
            else if (TextBox1.Text == "Debug_Options_Help")
            {
                RichTextBox1.Text = "Debug_Options_Custom";
                TextBox1.Text = "";
            }
            else if (TextBox1.Text == "Debug_AlerteLog_ForceCrash")
            {
                Application.Exit();
            }
            else if (TextBox1.Text == "Debug_AlerteLog")
            {
                alertelog alertelog = new alertelog();
                alertelog.Show();
                TextBox1.Text = "";
            }
            else if (TextBox1.Text == "Debug_Help")
            {
                RichTextBox1.Text = 
                    "Debug_AlerteLog\n" +
                    "Debug_App_ForceCrash\n" +
                    "Debug_Options\n" +
                    "Debug_ClearMemory";
                TextBox1.Text = "";
            }
            else if (TextBox1.Text == "Debug_App_ForceCrash")
            {
                Application.ExitThread();
            }
            else if (TextBox1.Text == "Debug_ClearMemory")
            {
                GC.Collect();
                GC.WaitForPendingFinalizers();
                RichTextBox1.Text = "[DEBUG]: Memory Cleared!";
                TextBox1.Text = "";
            }
            else
            {
                RichTextBox1.Text = "Commande non reconnu.";
                TextBox1.Text = "";
            }
        }
    }
}